client/CPlayerInterface.cpp

```diff
@@ -212,7 +213,15 @@ void CPlayerInterface::performAutosave()
 			prefix = settings["general"]["savePrefix"].String();
 			if(prefix.empty())
 			{
-				prefix = cb->getMapHeader()->name.substr(0, 5) + "_";
+				std::string name = cb->getMapHeader()->name;
+				std::wstring_convert<std::codecvt_utf8<char32_t>, char32_t> conv;
+				int txtlen = conv.from_bytes(name).size(); 

```

Comment:
Can you convert these two lines into new function in TextOperations namespaces,
e.g. 
`size_t DLL_LINKAGE getUnicodeCharactersCount(const std::string & text);`